### ISSUE TITLE
Broken Syntax Highlighting

### ISSUE THREAD

**Issue by Neo23x0** [infosec-intern/vscode-yara#47]
The syntax highlighting is broken for me and others since a few days. 
Could you please check? 

it broke with version 1.6.2

![Screenshot 2021-07-08 at 15 11 40](https://user-images.githubusercontent.com/2851492/125048920-b3270480-e0a0-11eb-96e5-883b0a1caf00.png)
![Screenshot 2021-07-08 at 15 11 57](https://user-images.githubusercontent.com/2851492/125048930-b4f0c800-e0a0-11eb-90df-fc36141402c4.png)

**Comment by DissectMalware** [comment]
```    
     {
      "name": "string.quoted.double.yara",
      "begin": "\"",
      "end": "[^\\\\]\"",
      "patterns": [
        {
          "name": "invalid.illegal.missing.escape.yara",
          "match": "[^\\\\]\\\\[^\\\\\"]"
        }
      ]
    }
```

changing [^\\\\]\" to (?<=[^\\])(\\((\\\\)*\\))?\" should solve the issue

zero or even numbers of backslash chars -> valid
odd numbers of backslash chars -> invalid

```
\\\\" -> valid
\\\\\" invalid
```

https://github.com/infosec-intern/vscode-yara/blob/fe196db7b914cbfe4e36838314dea2150984a7f9/yara/syntaxes/yara.tmLanguage.json#L148

**Comment by DissectMalware** [comment]
Changed  the following file

%USERPROFILE%\.vscode\extensions\infosec-intern.yara-1.6.4\yara\syntaxes\yara.tmLanguage.json

```
    {
      "name": "string.quoted.double.yara",
      "begin": "\"",
      "end": "(?<=[^\\\\])(\\\\((\\\\\\\\)*\\\\))?\"",
      "patterns": [
        {
          "name": "invalid.illegal.missing.escape.yara",
          "match": "[^\\\\]\\\\[^\\\\\"]"
        }
      ]
    }
```

After saving the change and reopening the VS Code, you should see:

![image](https://user-images.githubusercontent.com/9646319/125510141-09e5eabe-feb3-46bf-b4f0-b7c444194509.png)

**Comment by infosec-intern** [comment]
Dang, I really thought I had this sewed up in that version. Thanks for opening the issue @Neo23x0 and the writeup @DissectMalware! I'll get to fixing this later today

**Comment by infosec-intern** [comment]
Working on pushing out the fix in 1.6.5 now. This did uncover another issue with improper identification of invalid escaped slashes too, which I'll try to fix & test more comprehensively. No need for that to interrupt this though. Great work @DissectMalware, and thanks again for bringing the issue up @Neo23x0. If you find more of these please don't hesitate to bring them up

**Comment by infosec-intern** [comment]
This looks fixed in version [1.6.5](https://github.com/infosec-intern/vscode-yara/releases/tag/v1.6.5), which is now available in the marketplace. I'm gonna close this issue, but if this problem persists please feel free to reopen this or create a new issue.
